### ISSUE TITLE
Resume download after Instagram block 

### ISSUE THREAD

**Issue by plcdnl** [instaloader/instaloader#926]
Hi, all! I'm trying to download, from CLI, all photos of an hashtag that contains more of 70k post. After downloading about 1000/1200 photos, however, instagram blocks me. Is there a possibility to download a limit of photos a day without restart every time from start and always get stuck on the same photo?

**Comment by 2idiotz** [comment]
Did you found a solution? I'm also trying to get a list of all followers around 255k  but every time I stuck on 18k to 25k, after that it restarts from 0, no way to resume?

**Comment by plcdnl** [comment]
@2idiotz unfortunately, no ! I tried with --resume-prefix but it doesn't seem to work

**Comment by Nicopdev** [comment]
I'm having the same problem. I download around 25k usernames from a profile, but then the loader crashes because of `error 400 bad request` (which means that the account I am using to scrape data was blocked by instagram).
I can't find a way to tell instaloader that I want to get all the usernames starting from 25k the next time.
This feature might not be present in the current code, but I hope they will implement it asap.

**Comment by skeith** [comment]
Looks like what I am trying to achieve at #944

**Comment by Nicopdev** [comment]
I may have found a solution.
First of all, to avoid getting the account banned, my suggestion is to edit the default RateController.
Then you can use a try: except: block to download the data and save the iterator whenever the download fails.
Something like this:
`
try: 
    iterator = load_structure_from_file(file_dir) # Check if you have a frozen iterator in the file
    thaw(iterator.get_followers()) # If you do, resume that
except:
    iterator = target_profile.get_followers() # Else start the download from the first follower

try:
    # Iterate through the iterator
    for index, follower in enumerate(iterator):
        doSomethingWith(follower)
    except:
        save_structure_to_file(iterator.freeze(), file_dir) # Save the FrozenNodeIterator to a file whenever you get an exception (you may want to be more specific with your exception handling with something like “except BadRequestException:”
`

This works for me quite well, and I hope I was able to be clear enough on how to solve this issue

**Comment by 2idiotz** [comment]
@Nicopdev Can you verify if it resumes on next session after BadRequest(Temporary Block)?

**Comment by Nicopdev** [comment]
It does

**Comment by 2idiotz** [comment]
I'm curious if we can use another account to resume from last state :)

**Comment by Nicopdev** [comment]
> I'm curious if we can use another account to resume from last state :)

You should be able to check this by doing the following: after saving the file with the iterator inside of it, open it and edit everything that’s related to the current account to make it related to the next account.

To avoid the 400 bad request problem I also suggest to save the session of the login instead of logging in every time you start the loader

**Comment by 2idiotz** [comment]
Working!

**Comment by Nicopdev** [comment]
> Hi, all! I'm trying to download, from CLI, all photos of an hashtag that contains more of 70k post. After downloading about 1000/1200 photos, however, instagram blocks me. Is there a possibility to download a limit of photos a day without restart every time from start and always get stuck on the same photo?

Does this work for you too?

**Comment by subhead** [comment]
> I may have found a solution.
> First of all, to avoid getting the account banned, my suggestion is to edit the default RateController (this is “limit the number of pictures downloaded per day”)
> Then you can use a try: except: block to download the data and save the iterator whenever the download fails.
> Something like this: (sorry, can’t get code indentation to work as I am from my cellphone)
> 
> ```
> try:
>     iterator = load_structure_from_file(file_dir) # Check if you have a frozen iterator in the file
>     thaw(iterator.get_followers()) # If you do, resume that
> except:
>     iterator = target_profile.get_followers() # Else start the download from the first follower
> try:
>     # Iterate through the iterator
>     for index, follower in enumerate(iterator):
>         doSomethingWith(follower)
>     except:
>         save_structure_to_file(iterator.freeze(), file_dir) # Save the FrozenNodeIterator to a file whenever you get an exception (you may want to be more specific with your exception handling with something like “except BadRequestException:” 
> ```
> 
> This works for me quite well, and I hope I was able to be clear enough on how to solve this issue

Could you please explain this a little bit more? I dont know exactly what to edit.

**Comment by Nicopdev** [comment]
> > I may have found a solution.
> > First of all, to avoid getting the account banned, my suggestion is to edit the default RateController (this is “limit the number of pictures downloaded per day”)
> > Then you can use a try: except: block to download the data and save the iterator whenever the download fails.
> > Something like this: (sorry, can’t get code indentation to work as I am from my cellphone)
> > ```
> > try:
> >     iterator = load_structure_from_file(file_dir) # Check if you have a frozen iterator in the file
> >     thaw(iterator.get_followers()) # If you do, resume that
> > except:
> >     iterator = target_profile.get_followers() # Else start the download from the first follower
> > try:
> >     # Iterate through the iterator
> >     for index, follower in enumerate(iterator):
> >         doSomethingWith(follower)
> >     except:
> >         save_structure_to_file(iterator.freeze(), file_dir) # Save the FrozenNodeIterator to a file whenever you get an exception (you may want to be more specific with your exception handling with something like “except BadRequestException:” 
> > ```
> > 
> > 
> > This works for me quite well, and I hope I was able to be clear enough on how to solve this issue
> 
> Could you please explain this a little bit more? I dont know exactly what to edit.

Sure thing!
Is there any piece of code in particular that you don't understand?

**Comment by plcdnl** [comment]
> > Hi, all! I'm trying to download, from CLI, all photos of an hashtag that contains more of 70k post. After downloading about 1000/1200 photos, however, instagram blocks me. Is there a possibility to download a limit of photos a day without restart every time from start and always get stuck on the same photo?
> 
> Does this work for you too?

@Nicopdev Thank u for the interest. Unfortunately i haven't tried it yet. I'll write here when i do it

**Comment by Thammus** [comment]
Duplicate of #874.

**Comment by 2idiotz** [comment]
@Nicopdev I wanna see the code of imports, the thaw is not found.

**Comment by Nicopdev** [comment]
> @Nicopdev I wanna see the code of imports, the thaw is not found.

I’m sorry, that part of the code was not correct. It’s now updated

**Comment by ghalghay** [comment]
> 

Hi, Nicopdev, and soory my English! Could you give a complete code for python working with your nutrodies?
